writecrow/text_processing
# Remove non-printable characters, and remove 'NUL' character

## Purpose of changes
@gelan890926 identified texts which, after having been run through the `textnormalization.py` script, still retained characters that couldn't be interpreted for encoding.

The changes in this pull request add an additional method to remove non-printable characters (as defined by the [string library](https://docs.python.org/2/library/string.html#string.printable), as well as removing the [Null character](https://en.wikipedia.org/wiki/Null_character), which was ultimately causing the problem that Ge had encountered.

This pull request adds a test file at `normalization/test-data/normalization-1.txt` that illustrates the problem and demonstrates the fix.

## Testing Steps
0. Check out this branch's code (e.g., `git fetch && git checkout 12-nul-character`)
1. `cd normalization`
2. Check the file info/encoding on `test-data/normalization-1.txt`. It should be 'binary'.
3. `python textnormalization.py test-data/normalization-1.txt`
4. Check the file info/encoding on `cleaned/test-data/normalization-1.txt` (the output file). It should be us-ascii.
